pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/ExpressionType.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.vf;
+
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Logger;
+
+import apex.jorje.semantic.symbol.type.BasicType;
+import com.google.common.base.Strings;
+import com.google.common.collect.Sets;
+
+/**
+ * Represents all data types that can be referenced from a Visualforce page. This enum consolidates the data types
+ * available to CustomFields and Apex. It uses the naming convention of CustomFields.
+ *
+ * See https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_field_types.htm#meta_type_fieldtype
+ */
+public enum ExpressionType {
```

Comment:
It occurs to me that this should be renamed to `IdentifierType`. I will rename it.